### PR TITLE
Make dynamo tests work under pytest

### PR DESCRIPTION
**Pull request by jansel**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #93409
* #93408
* #93389
* #93382
* #93381
* #93279
* #93278
* #93375
* #93254
* #93253
* #93252
* __->__ #93251

This now runs without error:
```
pytest test/dynamo
```

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

**Review by ezyang — APPROVED** [review]

**Review by voznesenskym — APPROVED** [review]

File: test/dynamo/test_optimizers.py [hunk 1]
```diff
@@ -48,7 +48,9 @@ class OptimizerTests(torch._dynamo.test_case.TestCase):
     # furthermore, the break is inside a for loop, so we bail on the frame
     # entirely.  This is basically an xfail; if the frame count goes up
     # you done good
-    test_radam = make_test(torch.optim.RAdam, exp_graph_count=0)
+    test_radam = torch._dynamo.testing.skip_if_pytest(
```
Comment (mlazos) [inline test/dynamo/test_optimizers.py:51]: Curios why? There are some tests which are generated below which may need this as well

**Review by mlazos — APPROVED** [review]

File: test/dynamo/test_optimizers.py (hunk 1 above)
Comment (jansel) [reply test/dynamo/test_optimizers.py:51]: I think pytest rewrites python asserts at the bytecode level which triggers an extra graph break.  Didn't look into it in detail.

**Pull request closed without merging**